packages/backend-api/src/Controller/V1/Product/ProductController.php

```diff
@@ -0,0 +1,285 @@
+    {
+        $page = (int)$paramFetcher->get('page');
+
+        $query = new ProductQueryParams($this->pageSize, $page);
+
+        $filterUuids = $paramFetcher->get('uuids');
+        if (is_array($filterUuids)) {
+            $this->validateUuids($filterUuids);
+
+            $query = $query->withUuids($filterUuids);
+        }
+
+        $productsResult = $this->productFacade->findByProductQueryParams($query);
+
+        if ($page > $productsResult->getPage() || $page < 0) {
+            throw new UnprocessableEntityHttpException('There are no products on provided page.');
+        }
+
+        $productsArray = array_map(function (Product $product) {
+            return $this->productTransformer->transform($product);
+        }, $productsResult->getResults());
+
+        $links = $this->linksTransformer->fromPaginationResult($productsResult, $request->getUri());
+
+        $view = View::create($productsArray, Response::HTTP_OK, ['Link' => $links->format()]);
+
+        return $this->handleView($view);
+    }
+
+    /**
+     * @Post("/products")
+     * @Put("/products/{uuid}")
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @param string|null $uuid
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function createProductAction(Request $request, ?string $uuid = null): Response
+    {
+        $errors = $this->validateCreateProductInputs($request, $uuid);
+
+        if (count($errors) > 0) {
+            return $this->handleView($this->createValidationView($errors));
+        }
+
+        $productData = $this->productDataFactory->createFromApi($request->request->all(), $uuid);
+        $this->productFacade->create($productData);
+
+        $location = sprintf('%s/%s', $request->getUri(), $productData->uuid);
+        $view = View::create([], Response::HTTP_CREATED, ['Location' => $location]);
+
+        return $this->handleView($view);
+    }
+
+    /**
+     * @Delete("/products/{uuid}")
+     * @param string $uuid
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function deleteProductAction(string $uuid): Response
+    {
+        $this->validateUuids([$uuid]);
+
+        $product = $this->productFacade->getByUuid($uuid);
+        $this->disallowWorkingWithVariants($product);
+
+        $this->productFacade->delete($product->getId());
+
+        $view = View::create([], Response::HTTP_NO_CONTENT);
+
+        return $this->handleView($view);
+    }
+
+    /**
+     * @param array $errors
+     * @return \FOS\RestBundle\View\View
+     */
+    protected function createValidationView(array $errors): View
+    {
+        $code = Response::HTTP_BAD_REQUEST;
+        $message = 'Provided data did not pass validation';
+
+        return View::create(['code' => $code, 'message' => $message, 'errors' => $errors], $code);
+    }
+
+    /**
+     * @param array $uuids
+     */
+    protected function validateUuids(array $uuids): void
```

Comment:
I would like to see this method elsewhere, maybe `Validation` namespace so it could be reused